pkg/lwip/contrib/sock/lwip_sock.c

```diff
@@ -567,6 +567,7 @@ ssize_t lwip_sock_send(struct netconn **conn, const void *data, size_t len,
             break;
     }
     netbuf_delete(buf);
+    netconn_delete(tmp);
```

Comment:
This only should be deleted if it was created within this function, not in every case.

Reply:
You are absolutely right @miri64 ! I was already working on it :)
I made a flag _noconn_ that keeps track of whether a conn was passed or not. It's probably not the cleanest of solutions, but it seems like it does fix the issue for now.

@benpicco: Yes, my bad. I should have tested it more extensively. Should be fixed now. I tested on all 4 test/lwip's and all give a 100% pass.